solr/solrj/src/java/org/apache/solr/client/solrj/SolrRequest.java

```diff
@@ -260,18 +260,13 @@ public final T process(SolrClient client) throws SolrServerException, IOExceptio
   public final CompletableFuture<T> processAsynchronously(SolrClient client, String collection) {
     final long startNanos = System.nanoTime();
     final CompletableFuture<NamedList<Object>> internalFuture = client.requestAsync(this, collection);
-    final CompletableFuture<T> apiFuture = new CompletableFuture<>();
-
-    internalFuture.whenComplete((result, error) -> {
-      if (!internalFuture.isCompletedExceptionally()) {
-        T res = createResponse(client);
-        res.setResponse(result);
-        long endNanos = System.nanoTime();
-        res.setElapsedTime(TimeUnit.NANOSECONDS.toMillis(endNanos - startNanos));
-        apiFuture.complete(res);
-      } else {
-        apiFuture.completeExceptionally(error);
-      }
+
+    final CompletableFuture<T> apiFuture = internalFuture.thenApply((result) -> {
```

Comment:
What if internalFuture completes exceptionally?

Reply:
If internalFuture completes exceptionally, apiFuture will complete exceptionally with a `CompletionException` (with internalFuture's exception as its cause). That being said, I don't really like the `CompletionException` wrapper around the actual exception, so I'll revert to using whenComplete instead of thenApply.

Reply:
If internalFuture completes exceptionally, apiFuture will complete exceptionally with a `CompletionException` (with internalFuture's exception as its cause). That being said, I think it's better to not have a CompletionException wrapper around the actual exception, so I'll revert to using whenComplete instead of thenApply.